src/bucket/BucketManager.h

```diff
@@ -70,6 +70,12 @@ class BucketManager : NonMovableOrCopyable
 
     static std::string const kLockFilename;
 
+    // NB: ideally, BucketManager should have no access to mApp, as it's too
+    // dangerous in the context of parallel application. BucketManager is quite
+    // bloated, with lots of legacy code, so to ensure safety, annotate all
+    // functions using mApp with `releaseAssert(threadIsMain())` and avoid
+    // accessing mApp in the background. Safety invariant: lock acquisition must
+    // always be LCL lock -> BucketManger lock, and never the other direction
```

Comment:
This is ok but I think the comment on the lock mBucketMutex, below, is out of date: it's no longer just for protecting the file accesses, right?

Reply:
mm, I didn't change the semantics mBucketMutex though, instead I added a new lock to LM which guards changes to LCL.